main.go

```diff
@@ -211,6 +221,18 @@ func main() {
 		ClientDisableCacheFor: []client.Object{
 			&v3.LicenseKey{},
 		},
+		// NetworkPolicy is served through the Tigera API Server, which currently restricts List and Watch
+		// operations on NetworkPolicy to a single tier only, specified via label or field selector. If no
+		// selector is specified, List and Watch return policies from the 'default' tier. The manager cache
+		// must therefore apply a selector to specify the tier that the operator currently reconciles policy
+		// within so that it can receive the expected resources for List and Watch. If the operator needs to
+		// reconcile policy within multiple tiers, the API Server should be updated to serve policy from all
+		// tiers that the user is authorized for.
+		NewCache: cache.BuilderWithOptions(cache.Options{
```

Comment:
I believe we have to specify it on the cache for this kind, else disable the cache for this kind

Within ComponentHandler.CreateOrUpdateOrDelete, Client.Get is called to determine whether we need to update the object in question. If the cache is not disabled for the GVK, then the Client returned by the controller runtime uses its cache  to retrieve the object, which is where we run into the issue: the cache is filled using an inital List and updated via Watch. It's those List and Watch requests that we need to specify the label selector on for NetworkPolicy, else the cache is only going to have NetworkPolicy from the default tier.

From what I could see, this appeared to be the cleanest approach to apply the label selector to that List and Watch used for the cache, but I'm open to suggestions! I might be missing something.

Reply:
@Brian-McM when you and I last spoke, I was leaning towards that as well. However, as I was chatting with Matt on timelines for delivering this feature, we came to the conclusion that it was more pragmatic to make the small change in the operator now for the current requirement rather than rework tier RBAC in the API server for a future requirement (multiple tiers in the operator). I updated the design doc at that time to make sure it was included in the design review

I'm curious what both you and @caseydavenport think is best? I tried my best to summarize the context above and in the code comment, but also happy to hop on a call if its easier that way

Reply:
Maybe this is outside of the scope of what we should be commenting in a PR (and outside of what I should say as I'm not really in this design anymore and many decisions have been made without me) but it didn't seem like that API change was going to take an extraordinarily large amount of time. I had thought that more less exactly what needed to be done was figured out with you, Rob, and I.

Personally I think that if it's not going to be done right now for this feature, it should be done shortly after (as I think this is the correct thing to do). Given this, I'd say do whatever is quickest and easiest to implement (and undo) here, as it will just be reverted later when the proper fix is in for the API server.

Reply:
@pasanw thanks for that detailed write up. I think I see the issue now, and I'm leaning towards agreeing with you that specifying the selector on the cache is the most pragmatic solution to this for now. 

The downside is that we won't be able to do any Calico policy watches / management outside of the allow-tigera tier, which I believe will be overly limiting in the near future, so a proper fix is likely in order shortly after this.

I don't know what that fix would be - it might be server-side, or perhaps we can re-jig the way the controller infrastructure works somehow to solve this client-side. So basically, I also agree with @Brian-McM that we should schedule the "right" fix for this very soon after this.